src/deploy/functions/prepare.ts

```diff
@@ -112,19 +112,32 @@ export async function prepare(
     const userEnvs = functionsEnv.loadUserEnvs(userEnvOpt);
     const envs = { ...userEnvs, ...firebaseEnvs };
     const wantBuild: build.Build = await runtimeDelegate.discoverBuild(runtimeConfig, firebaseEnvs);
-    const wantBackend: backend.Backend = await build.resolveBackend(
+    const { backend: wantBackend, envs: resolvedEnvs } = await build.resolveBackend(
       wantBuild,
       userEnvOpt,
       userEnvs,
       options.nonInteractive
     );
+
+    let hasEnvsFromParams = false;
```

Comment:
see L175 - you might wan to add tracking to see how much usage param is getting.